### PR TITLE
added agnews dataset

### PR DESCRIPTION
**Pull request by bentrevett**
This is a very rough (but working) pull request that adds the AGNews dataset to TorchText. 

I know this is currently not up to scratch, but not really sure how to organise the code to make it sit nicely with the library so I'm mainly posting this to get feedback so I can go away and make the changes.

Once this has been all fixed and approved I want to move on to adding the rest of the datasets from [Character-level Convolutional Networks for Text Classification
](https://arxiv.org/abs/1509.01626), which are all available from the google drive link [here](https://github.com/zhangxiangxiao/Crepe).

All of the files added to `utils.py` were taken from TorchVision.

### CODE REVIEW

File: torchtext/utils.py [hunk 1]
```diff
@@ -75,3 +77,81 @@ def unicode_csv_reader(unicode_csv_data, **kwargs):
...
+
+def _get_confirm_token(response):
+    for key, value in response.cookies.items():
+        if key.startswith('download_warning'):
+            return value
+
+    return None
+
+def _save_response_content(response, destination, chunk_size=32768):
+    with open(destination, "wb") as f:
```
Comment (nzw0301) [inline torchtext/utils.py]: Could you use `'wb'` for consistency?

File: torchtext/utils.py [hunk 2]
```diff
@@ -75,3 +77,81 @@ def unicode_csv_reader(unicode_csv_data, **kwargs):
...
+    """Download a Google Drive file from  and place it in root.
+    Args:
+        file_id (str): id of file to be downloaded
+        root (str): Directory to place downloaded file in
+        filename (str, optional): Name to save the file under. If None, use the id of the file.
+        md5 (str, optional): MD5 checksum of the download. If None, do not check
+    """
+    # Based on https://stackoverflow.com/questions/38511444/python-download-files-from-google-drive-using-url
+    import requests
+    url = "https://docs.google.com/uc?export=download"
```
Comment (nzw0301) [inline torchtext/utils.py]: Could you use 'wb' for consistency?

File: torchtext/utils.py [hunk 3]
```diff
@@ -75,3 +77,81 @@ def unicode_csv_reader(unicode_csv_data, **kwargs):
...
+        return True
+    if not os.path.isfile(fpath):
+        return False
+    md5o = hashlib.md5()
+    with open(fpath, 'rb') as f:
+        # read in 1MB chunks
+        for chunk in iter(lambda: f.read(1024 * 1024), b''):
+            md5o.update(chunk)
+    md5c = md5o.hexdigest()
+    if md5c != md5:
```
Comment (nzw0301) [inline torchtext/utils.py]: How about `return md5c != md5` ?

File: torchtext/datasets/agnews.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    @staticmethod
+    def sort_key(ex):
+        return len(ex.description)
+
+    def __init__(self, path, title_field, description_field, label_field, **kwargs):
+        
+        fields = [('label', label_field), ('title', title_field), ('description', description_field)]
+        
+        with io.open(os.path.expanduser(path), encoding="utf8") as f:
```
Comment (nzw0301) [inline torchtext/datasets/agnews.py]: Could you use ' for consistency?

File: torchtext/utils.py (hunk 1 above)
Comment (bentrevett) [reply torchtext/utils.py]: done

File: torchtext/utils.py (hunk 2 above)
Comment (bentrevett) [reply torchtext/utils.py]: done

File: torchtext/utils.py (hunk 3 above)
Comment (bentrevett) [reply torchtext/utils.py]: done

File: torchtext/datasets/agnews.py (hunk 4 above)
Comment (bentrevett) [reply torchtext/datasets/agnews.py]: done

**Comment by erip** [comment]
I think this is safe to close since we have AG_NEWS now.

**Pull request closed without merging**